src/admin/class-datalayers.php

```diff
@@ -417,30 +496,48 @@ public static function override_source_fields_meta_value( $value, $object_id ) {
 			return $value;
 		}
 		$data = get_post_meta( $object_id, 'source_fields', true );
+
 		if ( $data ) {
 			return $value;
 		}
 
-		$datalayer_file = get_post_meta( $object_id, 'datalayer_file_id', true );
-		if ( ! $datalayer_file ) {
-			return $value;
-		}
-		$file = get_attached_file( $datalayer_file );
-		// phpcs:ignore WordPress.WP.AlternativeFunctions.file_get_contents_file_get_contents -- file_get_contents is allowed here.
-		$data = file_get_contents( $file );
-		$geom = geoPHP::load( $data );
-		$json = $geom->out( 'json' );
-		$data = json_decode( $json, true );
+		switch ( self::$datalayer_type ) {
+			case 'fileinput':
+			default:
+				$datalayer_file = get_post_meta( $object_id, 'datalayer_file_id', true );
+				if ( ! $datalayer_file ) {
+					return $value;
+				}
+				$file = get_attached_file( $datalayer_file );
+				// phpcs:ignore WordPress.WP.AlternativeFunctions.file_get_contents_file_get_contents -- file_get_contents is allowed here.
+				$json = file_get_contents( $file );
+				$data = json_decode( $json, true );
 
-		if ( ! $data ) {
-			return $value;
+				if ( ! $data ) {
+					return $value;
+				}
+
+				$data_array = $data['features'][0]['properties'];
+				break;
+			case 'url':
+				$url_data = self::fetch_datalayer_url_data( $object_id, true );
+
+				if ( empty( $url_data ) ) {
+					return $value;
+				}
+
+				$data_array = $url_data;
+				break;
 		}
 
 		$repeater = [];
-		foreach ( $data['features'][0]['properties'] as $key => $val ) {
+
+		foreach ( $data_array as $key => $val ) {
+			$display_label = $key;
+
 			$repeater[] = [
 				'field_label'         => $key,
-				'field_display_label' => $key,
+				'field_display_label' => $display_label,
```

Comment:
Waarom de tussenstap met assignment $display_labe = $key? Waarom niet direct 'field_display_label' => $key, zoals het was?

Reply:
Ik weet niet meer waarom ik dit gedaan heb, maar is inderdaad overbodig.